patches/minecraft/net/minecraft/client/multiplayer/PlayerControllerMP.java.patch

```diff
@@ -95,7 +95,17 @@
                      this.field_78776_a.func_147118_V().func_147682_a(new PositionedSoundRecord(soundtype.func_185846_f(), SoundCategory.NEUTRAL, (soundtype.func_185843_a() + 1.0F) / 8.0F, soundtype.func_185847_b() * 0.5F, p_180512_1_));
                  }
  
-@@ -350,7 +361,7 @@
+@@ -326,7 +337,8 @@
+ 
+     public float func_78757_d()
+     {
+-        return this.field_78779_k.func_77145_d() ? 5.0F : 4.5F;
++        float attrib = (float) field_78776_a.field_71439_g.func_110148_a(EntityPlayer.REACH_DISTANCE).func_111126_e();
++        return this.field_78779_k.func_77145_d() ? attrib : attrib - 0.5F;
```

Comment:
Not sure why vanilla logic has the server report reach dist 5 regardless of gamemode, but has the client shorten it by 0.5 in non-creative (perhaps to handle desyncs?). Either way, the logic is preserved.